src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackagesErrorBar.xaml.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+using Task = System.Threading.Tasks.Task;
+
+namespace NuGet.PackageManagement.UI
+{
+    [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
+    public partial class PackagesErrorBar : UserControl, INuGetProjectContext
+    {
+        private readonly ISolutionManager _solutionManager;
+
+        public event EventHandler RefreshUI;
+
+        public event EventHandler InitializationCompleted;
+
+        public PackageExtractionContext PackageExtractionContext { get; set; }
+
+        public ISourceControlManagerProvider SourceControlManagerProvider { get; }
+
+        public ProjectManagement.ExecutionContext ExecutionContext { get; }
+
+        public XDocument OriginalPackagesConfig { get; set; }
+
+        public NuGetActionType ActionType { get; set; }
+
+        public TelemetryServiceHelper TelemetryService { get; set; }
+
+        public PackagesErrorBar(ISolutionManager solutionManager)
+        {
+            InitializeComponent();
+
+            _solutionManager = solutionManager;
+
+            // Start a task to check for all the project's packages file and accordingly update error bar.
+            // this would also be initlize VSSolutionManager first time, if it hasn't been initialized yet.
+            Task.Run(UpdateErrorBarAsync);
+
+            ErrorMessage.SetResourceReference(TextBlock.ForegroundProperty, VsBrushes.InfoTextKey);
+            ErrorBar.SetResourceReference(Border.BackgroundProperty, VsBrushes.InfoBackgroundKey);
+            ErrorBar.SetResourceReference(Border.BorderBrushProperty, VsBrushes.ActiveBorderKey);
+        }
+
+        private async Task UpdateErrorBarAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
InitializeComponent is completed before even scheduling this task. And this API has to be executed on UI thread since it tries to update xaml component which is only accessible on UI. ThrowIfNotOnUIThread  won't help that would just always throw. We need to switch to UI here.